docs/coding-guidelines.md

```diff
@@ -149,6 +149,42 @@ For example the following are correct:
 
 1. Adding TODOs is not recommended and should be avoided whenever possible. In cases in which they do get added each TODO needs to be linked to an issue. Use the full url to the GitHub issue.
 
+1. Do not having trailing whitespace at the end of lines.
```

Comment:
Should we maybe be sorting the rules by context? 

For example this one could be right next to: 

> 1. Do add a new line at the end of all files.

Reply:
On one hand yes, I think that makes sense. but there's a bunch of PR comments saying "coding guidelines number 10", which will no longer make sense if we change the order. Perhaps we need to make them headings, so they each get an anchor that can be linked to?

Reply:
But then each rule would need a succinct heading, which might be more difficult. 

I'm fine if we keep the ordering I guess. I'm also personally fine with changing the numbers every once in a while :D